File: VB/WindowsApplication1/Helper/MyUnboundGridHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid
Imports System.IO
Imports DevExpress.XtraLayout.Resizing
Imports WindowsApplication1.Helper

Namespace WindowsApplication1
	<Designer(GetType(MyDesigner))> _
	Public Class MyUnboundGridHelper
		Inherits Component

		Public Sub New()

		End Sub

		Private _SelectedGridControl As GridControl
		Public Property SelectedGridControl() As GridControl
			Get
				Return _SelectedGridControl
			End Get
			Set(ByVal value As GridControl)
				_SelectedGridControl = value
				OnSelectedGridControlChanged()
			End Set
		End Property

		Private Sub OnSelectedGridControlChanged()
			If SelectedGridControl IsNot Nothing Then
				SelectedGridControl.DataSource = GridData
			End If
		End Sub

		Private _GridData As DataTable

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Property GridData() As DataTable
			Get
				If _GridData Is Nothing Then
					_GridData = New DataTable()
				End If
					If File.Exists(Constants.XmlFileName) Then
						_GridData.Clear()
						_GridData.ReadXmlSchema(Constants.SchemaName)
						_GridData.ReadXml(Constants.XmlFileName)
					End If
				Return _GridData
			End Get
			Set(ByVal value As DataTable)
				_GridData = value
			End Set
		End Property

		Public Sub ShowDesigner()
			CType(New UnboundDataForm(Me), UnboundDataForm).ShowDialog()
			File.Delete(Constants.XmlFileName)
			GridData.WriteXmlSchema(Constants.SchemaName)
			GridData.WriteXml(Constants.XmlFileName)
			SelectedGridControl.MainView.PopulateColumns()
		End Sub

	End Class

End Namespace

File: CS/WindowsApplication1/Helper/MyUnboundGridHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid;
using System.IO;
using DevExpress.XtraLayout.Resizing;
using WindowsApplication1.Helper;

namespace WindowsApplication1
{
    [Designer(typeof(MyDesigner))]
    public class MyUnboundGridHelper : Component
    {

        public MyUnboundGridHelper()
        {

        }

        private GridControl _SelectedGridControl;
        public GridControl SelectedGridControl
        {
            get { return _SelectedGridControl; }
            set { _SelectedGridControl = value; OnSelectedGridControlChanged(); }
        }

        private void OnSelectedGridControlChanged()
        {
            if (SelectedGridControl != null)
                SelectedGridControl.DataSource = GridData;
        }

        private DataTable _GridData;

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public DataTable GridData
        {
            get
            {
                if
              (_GridData == null) _GridData = new DataTable();
                    if (File.Exists(Constants.XmlFileName))
                    {
                        _GridData.Clear();
                        _GridData.ReadXmlSchema(Constants.SchemaName);
                        _GridData.ReadXml(Constants.XmlFileName);
                    }
                return _GridData;
            }
            set { _GridData = value;}
        }

        public void ShowDesigner()
        {
            new UnboundDataForm(this).ShowDialog();
            File.Delete(Constants.XmlFileName);
            GridData.WriteXmlSchema(Constants.SchemaName);
            GridData.WriteXml(Constants.XmlFileName);
            SelectedGridControl.MainView.PopulateColumns();
        }

    }

}